webapi/src/main/scala/org/knora/webapi/slice/admin/api/service/ProjectsLegalInfoRestService.scala

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright © 2021 - 2025 Swiss National Data and Service Center for the Humanities and/or DaSCH Service Platform contributors.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.knora.webapi.slice.admin.api.service
+
+import cats.syntax.traverse.*
+import org.eclipse.rdf4j.sparqlbuilder.rdf.Rdf
+import zio.IO
+import zio.Task
+import zio.ZIO
+import zio.ZLayer
+import zio.json.JsonCodec
+
+import dsp.errors.ForbiddenException
+import dsp.errors.InconsistentRepositoryDataException
+import dsp.errors.NotFoundException
+import org.knora.webapi.slice.admin.api.CopyrightHolderAddRequest
+import org.knora.webapi.slice.admin.api.CopyrightHolderReplaceRequest
+import org.knora.webapi.slice.admin.api.LicenseDto
+import org.knora.webapi.slice.admin.api.model.FilterAndOrder
+import org.knora.webapi.slice.admin.api.model.PageAndSize
+import org.knora.webapi.slice.admin.api.model.PagedResponse
+import org.knora.webapi.slice.admin.domain.model.Authorship
+import org.knora.webapi.slice.admin.domain.model.CopyrightHolder
+import org.knora.webapi.slice.admin.domain.model.KnoraProject
+import org.knora.webapi.slice.admin.domain.model.KnoraProject.Shortcode
+import org.knora.webapi.slice.admin.domain.model.User
+import org.knora.webapi.slice.admin.domain.service.KnoraProjectService
+import org.knora.webapi.slice.admin.domain.service.LicenseService
+import org.knora.webapi.slice.admin.domain.service.ProjectService
+import org.knora.webapi.slice.common.api.AuthorizationRestService
+import org.knora.webapi.slice.common.repo.rdf.Vocabulary
+import org.knora.webapi.store.triplestore.api.TriplestoreService
+import org.knora.webapi.store.triplestore.api.TriplestoreService.Queries.Select
+
+final case class ProjectsLegalInfoRestService(
+  private val licenses: LicenseService,
+  private val projects: KnoraProjectService,
+  private val auth: AuthorizationRestService,
+  private val triplestore: TriplestoreService,
+) {
+
+  def findAuthorships(user: User)(
+    shortcode: Shortcode,
+    pageAndSize: PageAndSize,
+    filterAndOrder: FilterAndOrder,
+  ): Task[PagedResponse[Authorship]] =
+    for {
+      project         <- auth.ensureProjectMember(user, shortcode)
+      totalAndAuthors <- queryAuthorships(project, pageAndSize, filterAndOrder)
+      (total, authors) = totalAndAuthors
+    } yield PagedResponse.from(authors, total, pageAndSize)
+
+  private def queryAuthorships(
+    project: KnoraProject,
+    paging: PageAndSize,
+    filterAndOrder: FilterAndOrder,
+  ): Task[(Int, Seq[Authorship])] = {
```

Comment:
Question: I did not expect to see database interaction in the RestService, thinking back to [this RFC](https://www.notion.so/dasch-swiss/draft-RFC-001-Use-Ports-and-Adapters-Architecture-Package-proposal-56eee56e3e6e4d569ace273e611b696b?pvs=4), I would have assumed there to be a Service and Repo in-between. Do you plan on leaving it like this or is this something you'd change eventually?

Reply:
Good point.

Moved it to the `LegalInfoService` => 832cd9b20b470a8f528551e1740585e8e1659de6